### PR TITLE
Add documentation for using UPM with MRTK

### PR DESCRIPTION
**Pull request by davidkline-ms**
This change contains new articles and updates to describe the MRTK UPM implementation. Also included are some new images to illustrate specific aspects of using MRTK via UPM.

### CODE REVIEW

File: Documentation/MRTK_PackageContents.md [hunk 1]
```diff
@@ -1,25 +1,40 @@
 # Mixed Reality Toolkit packages
 
-The Microsoft Mixed Reality Toolkit is provided as a collection of packages. The contents of these packages is described in the following sections.
+The Microsoft Mixed Reality Toolkit is provided as a collection of packages.
+
+- [Asset packages](#asset-packages)
+- [Unity Pacakge Manager](#unity-package-manager)
```
Comment (keveleigh) [inline Documentation/MRTK_PackageContents.md:6]:
```suggestion
- [Unity Package Manager](#unity-package-manager)
```

File: Documentation/MRTK_PackageContents.md [hunk 2]
```diff
@@ -70,7 +86,16 @@ These tools are located in the **Mixed Reality Toolkit > Utilities** menu in the
 | | ScreenshotUtility | Enables capturing application images in the Unity editor. |
 | | TextureCombinerWindow | Utility to combine graphics textures. |
 
-## Examples
+### Test Utiltiies
```
Comment (keveleigh) [inline Documentation/MRTK_PackageContents.md:89]:
```suggestion
### Test Utilities
```

File: Documentation/MRTK_PackageContents.md [hunk 3]
```diff
@@ -70,7 +86,16 @@ These tools are located in the **Mixed Reality Toolkit > Utilities** menu in the
...
+
+The optional Microsocft.MixedRealityToolkit,.TestUtilties package includes a collection of utilities used by the Mixed Reality team to author unit and other validation tests.
```
Comment (keveleigh) [inline Documentation/MRTK_PackageContents.md:91]:
```suggestion
The optional Microsocft.MixedRealityToolkit,.TestUtilities package includes a collection of utilities used by the Mixed Reality team to author unit and other validation tests.
```

File: Documentation/MRTK_PackageContents.md [hunk 4]
```diff
@@ -70,7 +86,16 @@ These tools are located in the **Mixed Reality Toolkit > Utilities** menu in the
...
+
+| Folder | Component | Description |
+| --- | --- | --- |
+| MRTK/Tests | |
+| | TestUtilities | Methods to simplify creation of play mode tests, including hand simiulation utilities. |
```
Comment (keveleigh) [inline Documentation/MRTK_PackageContents.md:96]:
```suggestion
| | TestUtilities | Methods to simplify creation of play mode tests, including hand simulation utilities. |
```

File: Documentation/MRTK_PackageContents.md [hunk 5]
```diff
@@ -85,6 +110,66 @@ The optional Microsoft.MixedRealityToolkit.Unity.Examples package includes demon
...
+- [Foundation](#foundation-upm)
+- [Standard Assets](#standard-assets-upm)
+- [Extensions](#extensions-upm)
+- [Tools](#tools-upm)
+- [Test Utilities](#test-utilties-upm)
+- [Examples](#examples-upm)
+
+### Foundation (UPM)
+
+The Unity Package Manager version of the MRTK foundation package (`com.microsoft.mixedreality.toolkit.foundation`) contains most of the same content as the [asset package version](#foundaton).
```
Comment (keveleigh) [inline Documentation/MRTK_PackageContents.md:126]:
```suggestion
The Unity Package Manager version of the MRTK foundation package (`com.microsoft.mixedreality.toolkit.foundation`) contains most of the same content as the [asset package version](#foundation). 
```

File: Documentation/MRTK_PackageContents.md [hunk 6]
```diff
@@ -85,6 +110,66 @@ The optional Microsoft.MixedRealityToolkit.Unity.Examples package includes demon
...
+
+The tools UPM package (`com.microsoft.mixedreality.toolkit.tools`) contains the same content as the [tools asset package](#tools). 
+
+### Test Utilities (UPM)
+
+The test utlities UPM package (`com.microsoft.mixedreality.toolkit.testutilities`) contains the same content as the [test utiltiies asset package](#test-utilities). 
+
+### Examples (UPM)
+
+The examples UPM package (`com.microsoft.mixedreality.toolkit.examples`) contains the same content as the [examples asset package](#examples)y.
```
Comment (keveleigh) [inline Documentation/MRTK_PackageContents.md:168]:
```suggestion
The examples UPM package (`com.microsoft.mixedreality.toolkit.examples`) contains the same content as the [examples asset package](#examples). 
```

File: Documentation/Packaging/MRTK_Packages.md [hunk 7]
```diff
@@ -2,16 +2,48 @@
...
-- [Examples](#examples-package)
 - [Tools](#tools-package)
+- [Test utilties](#test-utitilies-package)
+- [Examples](#examples-package)
 
 These packages are released and supported by Microsoft from source code in the [mrtk_release](https://github.com/Microsoft/MixedRealityToolkit-Unity/tree/mrtk_release) branch on GitHub.
 
-## Foundation package
+> [!Note]
+> For a detailed listing of the contents of each package, please see << link to package contents page (make a reference to asset packages) >>
```
Comment (keveleigh) [inline Documentation/Packaging/MRTK_Packages.md:44]: ><< link to package contents page (make a reference to asset packages) >>

Is this missing?

File: Documentation/Packaging/MRTK_Packages.md [hunk 8]
```diff
@@ -2,16 +2,48 @@
...
+  - Risk of losing changes made to MRTK 
+- Importing the examples package typically means including all examples
+
+The available packages are:
 
 - [Foundation](#foundation-package)
 - [Extensions](#extensions-package)
-- [Examples](#examples-package)
 - [Tools](#tools-package)
+- [Test utilties](#test-utitilies-package)
```
Comment (keveleigh) [inline Documentation/Packaging/MRTK_Packages.md:38]:
```suggestion
- [Test utilities](#test-utilities-package)
```

File: Documentation/usingupm.md [hunk 9]
```diff
@@ -0,0 +1,77 @@
+# Mixed Reality Toolkit and Unity Package Manager
+
+Starting with version 2.5.0, the Microsoft Mixed Reality Toolkit is available using the Unity Package Manager (UPM), on Unity 2019.4 and newer. Using UPM, the MRTK gains dependency management and a simple update path, as will be described in the following sections.
+
+## Installing Mixed Reality features using the Unity Package Manager
```
Comment (keveleigh) [inline Documentation/usingupm.md:5]: Do we also want to upload the .tgz files to our releases, so people can download them and import to UPM via the [local tarball option](https://docs.unity3d.com/Manual/upm-ui-tarball.html)?

File: Documentation/usingupm.md [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+> [!Note]
+> If a Mixed Reality Toolkit package is removed using the Unity Package Manager, it will have to be re-added using the [previously described steps](#adding-mrtk-packages).
+
+### Using Mixed Reality Toolkit examples
+
+Packages installed via the Unity Package Manager are immutable. Scripts, scenes and other assets are read-only when installed via packages. Additionally, scenes that are contained within a UPM package cannot be selected in the Build Settings window.
+
+To allow for these scenarios, MRTK packages that contain sample projects (`com.microsoft.mixedreality.toolkit.examples` and `com.microsoft.mixedreality.toolkit.handphysicsservice`) have the ability to import specific samples into the Assets folder.
+
+![Importing samples](../Images/Packages/Examples_Samples.png)
```
Comment (keveleigh) [inline Documentation/usingupm.md:64]:
```suggestion
![Importing samples](../Images/Packaging/MRTK_ExamplesUpm.png)
```

File: Documentation/Packaging/MRTK_Packages.md [hunk 11]
```diff
@@ -68,17 +123,102 @@ This package also contains eye tracking demos, which are [documented here](../Ey
...
+
+Dependencies: none
+
+### Extension pacakges
+
+The extensions package (`com.microsoft.mixedreality.toolkit.extensions)` contains additional components that expand the functionality of the MRTK. 
+
+<< >>
+
+![Extensions samples](../Images/Packages/Extensions_Samples.png)
```
Comment (keveleigh) [inline Documentation/Packaging/MRTK_Packages.md:188]:
```suggestion
![Extensions samples](../Images/Packaging/Extensions_Samples.png)
```

File: Documentation/usingupm.md [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+> [!Note]
+> If a Mixed Reality Toolkit package is removed using the Unity Package Manager, it will have to be re-added using the [previously described steps](#adding-mrtk-packages).
+
+### Using Mixed Reality Toolkit examples
+
+Packages installed via the Unity Package Manager are immutable. Scripts, scenes and other assets are read-only when installed via packages. Additionally, scenes that are contained within a UPM package cannot be selected in the Build Settings window.
+
+To allow for these scenarios, MRTK packages that contain sample projects (`com.microsoft.mixedreality.toolkit.examples` and `com.microsoft.mixedreality.toolkit.handphysicsservice`) have the ability to import specific samples into the Assets folder.
+
+![Importing samples](../Images/Packaging/MRTK_ExamplesUpm.png)
```
Comment (keveleigh) [inline Documentation/usingupm.md:64]:
```suggestion
![Importing samples](Images/Packaging/MRTK_ExamplesUpm.png)
```

File: Documentation/Packaging/MRTK_Packages.md (hunk 7 above)
Comment (davidkline-ms) [reply Documentation/Packaging/MRTK_Packages.md:44]: yep. adding it shortly

File: Documentation/usingupm.md (hunk 9 above)
Comment (davidkline-ms) [reply Documentation/usingupm.md:5]: possibly

File: Documentation/Packaging/MRTK_Packages.md (hunk 7 above)
Comment (davidkline-ms) [reply Documentation/Packaging/MRTK_Packages.md]: fixed

File: Documentation/usingupm.md (hunk 9 above)
Comment (davidkline-ms) [reply Documentation/usingupm.md:5]: let's add those instructions when we decide if that is the route we decide to take

File: Documentation/MRTK_PackageContents.md [hunk 13]
```diff
@@ -85,6 +110,66 @@ The optional Microsoft.MixedRealityToolkit.Unity.Examples package includes demon
...
+### Test Utilities (UPM)
+
+The test utlities UPM package (`com.microsoft.mixedreality.toolkit.testutilities`) contains the same content as the [test utiltiies asset package](#test-utilities). 
+
+### Examples (UPM)
+
+The examples UPM package (`com.microsoft.mixedreality.toolkit.examples`) contains the same content as the [examples asset package](#examples). 
+
+> [!Note]
+> When importing the examples UPM package, sample scenes are not imported by default.
```
Comment (thalbern) [inline Documentation/MRTK_PackageContents.md:171]: do you think we should make this clearer by writing adding something like: When importing the examples UPM package, sample scenes are not imported by default and need to be explicitely enabled in the import dialog.

just to avoid confusion that scenes might not be part of the package in UPM?

File: Documentation/Packaging/MRTK_Packages.md [hunk 14]
```diff
@@ -20,43 +52,66 @@ The Mixed Reality Toolkit Foundation is the set of code that enables your applic
...
 - [Scene Transition Service](../Extensions/SceneTransitionService/SceneTransitionServiceOverview.md)
 
-## Examples package
+### Tools package
+
+The tools package contains tools that are useful for creating mixed reality experiences. In general, these tools are editor components and their code does not ship as part of an application.
+
+Some of the tools in this package include:
+
+- [Dependency Window](../Tools/DependencyWindow.md)
```
Comment (thalbern) [inline Documentation/Packaging/MRTK_Packages.md:105]: should we mention the migration window as well?

File: Documentation/usingupm.md [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+    }
+  ],
+  "dependencies": {
+  ```
+
+### Adding MRTK packages
+
+Once the Microsoft Mixed Reality scoped registry has been added to the manifest, the MRTK packages can be specified.
+
+The [Unity Package Manager](Packaging/MRTK_Packages.md#unity-package-manager) section of the [Mixed Reality Toolkit package](Packaging/MRTK_Packages.md) article describes the avaiable MRTK packages, their contents and the scenarios for thir use.
```
Comment (thalbern) [inline Documentation/usingupm.md:38]:
```suggestion
The [Unity Package Manager](Packaging/MRTK_Packages.md#unity-package-manager) section of the [Mixed Reality Toolkit package](Packaging/MRTK_Packages.md) article describes the avaiable MRTK packages, their contents and the scenarios for their use.
```

File: Documentation/usingupm.md [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+
+To add an MRTK package, modify the dependencies section of the `Packages/manifest.json` file. The following example illustrates adding the foundation, tools and examples packages, the standard assets package will be added automatically as a dependency of the foundation..
```
Comment (thalbern) [inline Documentation/usingupm.md:40]: there's a dot too much at the end of the sentence (and for some reason i can't remove via suggestion button :-/)

File: Documentation/usingupm.md [hunk 17]
```diff
@@ -0,0 +1,75 @@
...
+    "com.microsoft.mixedreality.toolkit.foundation": "2.5.0",
+    "com.microsoft.mixedreality.toolkit.tools": "2.5.0",
+    "com.microsoft.mixedreality.toolkit.examples": "2.5.0",
+```
+
+## Managing Mixed Reality features with the Unity Package Manager
+
+Once a Mixed Reality Toolkit package has been added to the package manifest, it can be managed using the Unity Package Manager user interface.
+
+<< image of MRTK package in the UPM UI >>
```
Comment (thalbern) [inline Documentation/usingupm.md:53]: adding comment here so we won't forget to add this picture :)

File: Documentation/usingupm.md [hunk 18]
```diff
@@ -0,0 +1,75 @@
...
+
+> [!Note]
+> If a Mixed Reality Toolkit package is removed using the Unity Package Manager, it will have to be re-added using the [previously described steps](#adding-mrtk-packages).
+
+### Using Mixed Reality Toolkit examples
+
+Packages installed via the Unity Package Manager are immutable. Scripts, scenes and other assets are read-only when installed via packages. Additionally, scenes that are contained within a UPM package cannot be selected in the Build Settings window.
```
Comment (thalbern) [inline Documentation/usingupm.md:60]: should we highlight the fact that example scenes out of the upm package can not be deployed to device?

File: Documentation/Packaging/MRTK_Packages.md (hunk 14 above)
Comment (davidkline-ms) [reply Documentation/Packaging/MRTK_Packages.md:105]: we can

File: Documentation/usingupm.md (hunk 16 above)
Comment (davidkline-ms) [reply Documentation/usingupm.md:40]: will fix!

File: Documentation/usingupm.md [hunk 19]
```diff
@@ -0,0 +1,74 @@
+# Mixed Reality Toolkit and Unity Package Manager
+
+Starting with version 2.5.0, the Microsoft Mixed Reality Toolkit is available using the Unity Package Manager (UPM), on Unity 2019.4 and newer. Using UPM, the MRTK gains dependency management and a simple update path, as will be described in the following sections.
```
Comment (wiwei) [inline Documentation/usingupm.md:3]:
```suggestion
Starting with version 2.5.0, the Microsoft Mixed Reality Toolkit is available using the Unity Package Manager (UPM), on Unity 2019.4 and newer.
```

Comment (wiwei) [reply Documentation/usingupm.md:3]: Since this is going to be a longer-running doc, this part doesn't really add a ton of value for folks after the fact.

File: Documentation/usingupm.md [hunk 20]
```diff
@@ -0,0 +1,74 @@
...
+
+## Installing Mixed Reality features using the Unity Package Manager
+
+The Unity Package Mananger uses a manifest file (manifest.json) to determine which packages to install and the registries (servers) from which they can be installed.
```
Comment (wiwei) [inline Documentation/usingupm.md:7]:
```suggestion
The Unity Package Mananger uses a [manifest file](https://docs.unity3d.com/Manual/upm-manifestPkg.html) (manifest.json) to determine which packages to install and the registries (servers) from which they can be installed.
```

File: Documentation/usingupm.md [hunk 21]
```diff
@@ -0,0 +1,74 @@
...
+
+> [!Note]
+> As of version 2.5.0 of the MRTK, initial registration of the server and packages is a per-project, manual procedure, please read the following sections for detailed instructions.
```
Comment (wiwei) [inline Documentation/usingupm.md:10]:
```suggestion
> As of version 2.5.0 of the MRTK, initial registration of the server and packages is a per-project, manual procedure. Please read the following sections for detailed instructions.
```

File: Documentation/usingupm.md [hunk 22]
```diff
@@ -0,0 +1,74 @@
...
+
+### Registering the Mixed Reality component server
+
+For each project that will be using the Microsoft Mixed Realuty Toolkit, the `manifest.json` file (in the Packages folder) will need to have the Mixed Reality scoped registry added. The following illustrate how to properly modify `manfest.json` to support Mixed Reality.
```
Comment (wiwei) [inline Documentation/usingupm.md:14]: Realuty?!?!?

Can you run these changes using this:

https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker

I think there were a bunch of other misspellings in the other file, would be great if you could run this to avoid having reviewers do a manual spell checking pass.

File: Documentation/usingupm.md [hunk 23]
```diff
@@ -0,0 +1,74 @@
...
+    {
+      "name": "Microsoft Mixed Reality",
+      "url": "https://pkgs.dev.azure.com/aipmr/MixedReality-Unity-Packages/_packaging/Unity-packages/npm/registry/",
+      "scopes": [
+        "com.microsoft.mixedreality",
+        "com.microsoft.spatialaudio"
+      ]
+    }
+  ],
+  "dependencies": {
```
Comment (wiwei) [inline Documentation/usingupm.md:31]: Do you think people will copy paste this entire section and inadvertently copy the dependencies line as well?

File: Documentation/usingupm.md [hunk 24]
```diff
@@ -0,0 +1,74 @@
...
+  ```
+
+### Adding MRTK packages
+
+Once the Microsoft Mixed Reality scoped registry has been added to the manifest, the MRTK packages can be specified.
+
+The [Unity Package Manager](Packaging/MRTK_Packages.md#unity-package-manager) section of the [Mixed Reality Toolkit package](Packaging/MRTK_Packages.md) article describes the avaiable MRTK packages, their contents and the scenarios for their use.
+
+To add an MRTK package, modify the dependencies section of the `Packages/manifest.json` file. The following example illustrates adding the foundation, tools and examples packages, the standard assets package will be added automatically as a dependency of the foundation..
```
Comment (wiwei) [inline Documentation/usingupm.md:40]:
```suggestion
To add an MRTK package, modify the dependencies section of the `Packages/manifest.json` file. The following example illustrates adding the foundation, tools and examples packages, the standard assets package will be added automatically as a dependency of the foundation.
```

File: Documentation/usingupm.md [hunk 25]
```diff
@@ -0,0 +1,74 @@
...
+To add an MRTK package, modify the dependencies section of the `Packages/manifest.json` file. The following example illustrates adding the foundation, tools and examples packages, the standard assets package will be added automatically as a dependency of the foundation.
+
+```
+  "dependencies": {
+    "com.microsoft.mixedreality.toolkit.foundation": "2.5.0",
+    "com.microsoft.mixedreality.toolkit.tools": "2.5.0",
+    "com.microsoft.mixedreality.toolkit.examples": "2.5.0",
+```
+
+## Managing Mixed Reality features with the Unity Package Manager
```
Comment (wiwei) [inline Documentation/usingupm.md:49]: It might be worth having a section here about why there's this this... differing thing between adding it and removing it? i.e. the issues around UPM's built-in package management system w/ enumeration (using the deprecated ALL endpoint etc)

File: Documentation/MRTK_PackageContents.md [hunk 26]
```diff
@@ -1,25 +1,40 @@
 # Mixed Reality Toolkit packages
```
Comment (wiwei) [inline Documentation/MRTK_PackageContents.md:1]: Both of these Packages.md files seem to have a lot of overlap (indeed, both of them begin with the same title "Mixed Reality Toolkit packages".

Duplication is the bane of a lot of things - curious if you think it would be worth converging the two?

File: Documentation/Packaging/MRTK_Packages.md [hunk 27]
```diff
@@ -20,43 +52,66 @@ The Mixed Reality Toolkit Foundation is the set of code that enables your applic
...
+- [Extension Service Creation Wizard](../Tools/ExtensionServiceCreationWizard.md)
+- [Optimize Window](../Tools/OptimizeWindow.md)
+- [Screenshot Utility](../Tools/ScreenshotUtility.md)
+
+### Test utlities package
+
+The test utilities package contains a collection of helper scripts that enable developers to easily create play mode tests. These utilities are especially useful for developers creating MRTK components.
```
Comment (wiwei) [inline Documentation/Packaging/MRTK_Packages.md:112]: Nit: Consider adding a link to:

https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/Contributing/UnitTests.html#play-mode-tests

For examples on the types of test you could write with these utilities.

File: Documentation/Packaging/MRTK_Packages.md [hunk 28]
```diff
@@ -2,16 +2,48 @@
...
+
+
+## Asset packages
+
+The MRTK asset (.unitypackage) can be downloaded from [GitHub](https://github.com/microsoft/MixedRealityToolkit-Unity/releases). 
+
+Some of the benefits of using asset packages include:
+
+- Available for Unity 2018.4 and newer
+- Can configure which files are imported
```
Comment (wiwei) [inline Documentation/Packaging/MRTK_Packages.md:15]: Nit: It feels a bit misleading on this one - you definitely can omit things but it's notoriously hard for anyone who isn't us to do this. 

It can be hard to say it's a benefit in this case.

File: Documentation/usingupm.md (hunk 18 above)
Comment (davidkline-ms) [reply Documentation/usingupm.md:60]: they can. if you import them into the project. otherwise, they are "invisible" to the user (in the Samples~) folder.

Like things that start with '.', paths ending in '~' are not visible in the Unity editor

Comment (davidkline-ms) [reply Documentation/usingupm.md:60]: i will reword this to make it more clear

Comment (davidkline-ms) [reply Documentation/usingupm.md]: updated

File: Documentation/usingupm.md (hunk 17 above)
Comment (davidkline-ms) [reply Documentation/usingupm.md]: added

File: Documentation/usingupm.md (hunk 16 above)
Comment (davidkline-ms) [reply Documentation/usingupm.md]: fixed

File: Documentation/usingupm.md (hunk 22 above)
Comment (davidkline-ms) [reply Documentation/usingupm.md:14]: haha! typing too fast 😺

Comment (davidkline-ms) [reply Documentation/usingupm.md:14]: fixed

File: Documentation/usingupm.md (hunk 23 above)
Comment (davidkline-ms) [reply Documentation/usingupm.md:31]: it's possible. i'll remove it

Comment (davidkline-ms) [reply Documentation/usingupm.md]: done

File: Documentation/toc.yml [hunk 29]
```diff
@@ -13,8 +13,8 @@
     href: BuildAndDeploy.md
   - name: MRTK and Managed Code Stripping
     href: MRTK_and_managed_code_stripping.md
-  - name: NuGet Packages
-    href: MRTKNuGetPackage.md
```
[the remark this answers is not in the document]
Comment (davidkline-ms) [reply Documentation/toc.yml:17]: it doesnt hurt to remove it

File: Documentation/MRTK_PackageContents.md (hunk 26 above)
Comment (davidkline-ms) [reply Documentation/MRTK_PackageContents.md:1]: i was thinking that earlier.... i'll fold them

File: Documentation/usingupm.md [hunk 30]
```diff
@@ -0,0 +1,77 @@
...
+1. In the list of packages, select `Mixed Reality Toolkit Examples`
+1. Locate the desired sample(s) in the `Samples` list
+1. Click `Import into Project`
+
+![Importing samples](Images/Packaging/MRTK_ExamplesUpm.png)
+
+When an example package is updated, Unity provides the option to update imported samples.
+
+> [!Note]
+> Updating an imported sample will overwrite any changes that have been made to that sample and it's assets.
```
Comment (MaxWang-MS) [inline Documentation/usingupm.md:75]: I think you might have meant to say "its assets" instead of "it's assets"?

File: Documentation/usingupm.md (hunk 25 above)
Comment (davidkline-ms) [reply Documentation/usingupm.md:50]: added
